packages/survey-creator-core/src/components/action-container-view-model.ts

```diff
@@ -4,13 +4,75 @@ import {
   Action,
   SurveyModel,
   SurveyElement,
-  property
+  property,
+  actionModeType
 } from "survey-core";
 import { CreatorBase } from "../creator-base";
 import { settings } from "../creator-settings";
 
+export class SurveyElementActionContainer extends AdaptiveActionContainer {
+  private setActionMode(actionId: string, mode: actionModeType) {
+    const action = this.getActionById(actionId);
+    if (action) action.mode = mode;
+  }
+  public fit(dimension: number, dotsItemSize: number) {
+    if (dimension <= 0) return;
+
+    this.dotsItem.visible = false;
+    const items = this.visibleActions;
+
+    let size = 0;
+    items.forEach(i => size += i.maxDimension);
+    if (dimension >= size) {
+      // all items large
+      items.forEach(i => i.mode = "large");
+      return;
+    }
+
+    const actionsExceptInputType = items.filter(i => i.id != "convertInputType");
+    const actionsExceptTypeAndInputType = items.filter(i => i.id != "convertInputType" && i.id != "convertTo");
+
+    size = 0;
+    actionsExceptInputType.forEach(i => size += i.maxDimension);
+    if (dimension >= size) {
+      // all items large, input type removed
+      actionsExceptInputType.forEach(i => i.mode = "large");
+      this.setActionMode("convertInputType", "removed");
+      return;
+    }
+
+    size = 0;
+    actionsExceptTypeAndInputType.forEach(i => size += i.minDimension);
+    size += this.getActionById("convertTo")?.maxDimension;
+    if (dimension >= size) {
+      // all items small except question type
+      actionsExceptTypeAndInputType.forEach(i => i.mode = "small");
```

Comment:
Code duplication 
      actionsExceptTypeAndInputType.forEach(i => i.mode = actionMode);
      this.setActionMode("convertInputType", convertInputTypeMode);
      this.setActionMode("convertTo", convertToMode);

Reply:
I re-wrote this method